components/script/dom/gpudevice.rs

```diff
@@ -866,27 +494,11 @@ impl GPUDeviceMethods for GPUDevice {
         descriptor: &GPURenderPipelineDescriptor,
         comp: InRealm,
     ) -> Fallible<Rc<Promise>> {
-        let (implicit_ids, desc) = self.parse_render_pipeline(&descriptor)?;
-
         let promise = Promise::new_in_current_realm(comp);
+        // FIXME: if ``GPURenderPipeline::create` fails we should remove IpcReceiver from set
+        // but this is currently not possible
```

Comment:
[`IpcReceiverSet`](https://docs.rs/ipc-channel/latest/ipc_channel/ipc/struct.IpcReceiverSet.html) does not support removing. In webgpu we create new receiver for every `request_async` and they are only used once (sometimes twice but that's wrong).

Reply:
Why don't you move the call to `device.parse_render_pipeline` to here so you can fail before creating the ipc route?

Reply:
Good idea, will do but it would still be good idea to remove receivers that will not be used anymore.